demisto_sdk/common/hook_validations/release_notes.py

```diff
@@ -0,0 +1,152 @@
+from __future__ import print_function
+
+import os
+import re
+
+from demisto_sdk.common.tools import run_command
+from demisto_sdk.common.tools import print_error, get_latest_release_notes_text, \
+    get_release_notes_file_path
+
+
+class ReleaseNotesValidator:
+    """Release notes validator is designed to ensure the existence and correctness of the release notes in content repo.
+
+    Attributes:
+        file_path (str): the path to the file we are examining at the moment.
+        release_notes_path (str): the path to the changelog file of the examined file.
+        latest_release_notes (str): the text of the UNRELEASED section in the changelog file.
+        master_diff (str): the changes in the changelog file compared to origin/master.
+    """
+
+    LINK_TO_RELEASE_NOTES_STANDARD = 'https://github.com/demisto/content/blob/master/docs/release_notes/README.md'
+
+    def __init__(self, file_path):
+        self.file_path = file_path
+        self.release_notes_path = get_release_notes_file_path(self.file_path)
+        self.latest_release_notes = get_latest_release_notes_text(self.release_notes_path)
+        self.master_diff = self.get_master_diff()
+
+    def get_master_diff(self):
+        """Gets difference between current branch and origin/master
+
+        git diff with the --unified=100 option means that if there exists a
+        difference between origin/master and current branch, the output will have at most 100
+        lines of context.
+
+        Returns:
+            str. empty string if no changes made or no origin/master branch, otherwise full difference context.
+        """
+        return run_command(F'git diff --unified=100 '
+                           F'origin/master {self.release_notes_path}')  # type: str
+
+    def is_release_notes_changed(self):
+        """Validates that a new comment was added to release notes.
+
+        Returns:
+            bool. True if comment was added, False otherwise.
+        """
+        # there exists a difference between origin/master and current branch
+        if self.master_diff:
+            diff_releases = self.master_diff.split('##')
+            unreleased_section = diff_releases[1]
+            unreleased_section_lines = unreleased_section.split('\n')
+
+            adds_in_diff = 0
+            removes_in_diff = 0
+
+            for line in unreleased_section_lines:
+                if line.startswith('+'):
+                    adds_in_diff += 1
+                elif line.startswith('-') and not re.match(r'- *$', line):
+                    removes_in_diff += 1
+
+            # means that at least one new line was added
+            if adds_in_diff - removes_in_diff > 0:
+                return True
+
+        print_error(F'File {self.release_notes_path} is not formatted according to '
+                    F'release notes standards.\nFix according to {self.LINK_TO_RELEASE_NOTES_STANDARD}')
+        return False
```

Comment:
1. Why is that not inside the general "if" statement? we would like to print an error if there's no difference?
2. Let's improve the error, please write the exact problem to fix

Reply:
Done